pkg/k8shandler/collection.go

```diff
@@ -0,0 +1,166 @@
+package k8shandler
+
+import (
+  "k8s.io/apimachinery/pkg/api/errors"
+  "github.com/sirupsen/logrus"
+  "github.com/openshift/cluster-logging-operator/pkg/utils"
+  "k8s.io/api/core/v1"
+
+  sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
+  logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+)
+
+func CreateOrUpdateCollection(logging *logging.ClusterLogging) error {
+  createOrUpdateFluentdServiceAccount(logging)
+
+  createOrUpdateCollectionPriorityClass(logging)
+  createOrUpdateFluentdConfigMap(logging)
+  createOrUpdateFluentdDaemonset(logging)
+
+  return createOrUpdateFluentdSecret(logging)
+}
+
+func createOrUpdateCollectionPriorityClass(logging *logging.ClusterLogging) error {
+
+  collectionPriorityClass := utils.PriorityClass("cluster-logging", 1000000, false, "This priority class is for the Cluster-Logging Collector")
+
+  sdk.Create(collectionPriorityClass)
+  // Currently we cannot do validation checking on this due to version issues with the scheduling api and the version we create the priorityclass with
+  /*err := sdk.Create(collectionPriorityClass)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure creating Collection priority class: %v", err)
+  }*/
+
+  return nil
+}
+
+func createOrUpdateFluentdServiceAccount(logging *logging.ClusterLogging) error {
+
+  fluentdServiceAccount := utils.ServiceAccount("aggregated-logging-fluentd", logging.Namespace)
+
+  err := sdk.Create(fluentdServiceAccount)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure creating Fluentd service account: %v", err)
+  }
+
+  return nil
+}
+
+func createOrUpdateFluentdConfigMap(logging *logging.ClusterLogging) error {
+
+  // TODO: update to use the actual files we want...
+  fluentdConfigMap := utils.ConfigMap(
+    "logging-fluentd",
+    logging.Namespace,
+    map[string]string{
+      "fluent.conf": string(utils.GetFileContents("files/fluent.conf")),
+      "throttle-config.yaml": string(utils.GetFileContents("files/fluentd-throttle-config.yaml")),
+      "secure-forward.conf": string(utils.GetFileContents("files/secure-forward.conf")),
+    },
+  )
+
+  err := sdk.Create(fluentdConfigMap)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure constructing Fluentd configmap: %v", err)
+  }
+
+  return nil
+}
+
+func createOrUpdateFluentdSecret(logging *logging.ClusterLogging) error {
```

Comment:
createOrUpdateCollectorSecret  would this secret be different between fluent and rsyslog?

Reply:
potentially, yes. its just following the naming pattern of our other functions as we could be creating a secret for either fluentd or rsyslog